Classes/Core/Testbase.php

```diff
@@ -657,19 +653,47 @@ private function truncateAllTablesForMysql(): void
         /** @var Connection $connection */
         $connection = GeneralUtility::makeInstance(ConnectionPool::class)
             ->getConnectionByName(ConnectionPool::DEFAULT_CONNECTION_NAME);
-
         $databaseName = $connection->getDatabase();
-        $query = "SHOW TABLE STATUS FROM $databaseName";
+        $tableNames = $connection->getSchemaManager()->listTableNames();
+
+        if (empty($tableNames)) {
+            // No tables to process
+            return;
+        }
+
+        // Build a sub select to get joinable table with information if table has at least one row.
+        // This is needed because information_schema.table_rows is not reliable enough for innodb engine.
+        // see https://dev.mysql.com/doc/mysql-infoschema-excerpt/5.7/en/information-schema-tables-table.html TABLE_ROWS
+        $fromTableUnionSubSelectQuery = [];
+        foreach($tableNames as $tableName) {
+            $fromTableUnionSubSelectQuery[] = sprintf(
```

Comment:
I propose we do this only for tables that don't have an auto increment (to improve performance).

Reply:
To my measurements, this does not make much of a difference. Both the version before and with this patch seem to be equally fast. Looks as if it's fine to avoid additional logic to separate this, here. What do you think?

Reply:
If the performance is the same, then I'm fine with it. :-)